<?xml version="1.0" encoding="UTF-8"?>
    <root BTCPP_format="4" main_tree_to_execute="MainTree">
    <include path="takeoff.xml"/>
    <include path="planner_node.xml" />
    <include path="arrive_goal.xml" />
    <include path="land.xml" />
    <include path="hover.xml" />
    <include path="hit.xml" />
    <BehaviorTree ID="MainTree">
    <Sequence>
        <!-- <SetBlackboard name="goal" value="0.0;0.0;2.5" output_key="goal"/> -->
        <SetBlackboard name="is_rc" value="1" output_key="is_rc"/>
        <Script code=" P1_goal:='0.0;1.8;0.8' "/>  <!--P1点位置 -->
        <Script code=" P2_goal:='4.2;2.1;1.0' "/>  <!--P2点位置 -->
        <Script code=" P3_goal:='8;0;1.0' "/>  <!--P2点位置 -->
        <Script code=" goal_ori:='0;0.0;0' "/>  <!--P2目标点，飞机朝向 -->
        <SetBlackboard name="enabel_yaw" value="false" output_key="enabel_yaw" />
        <SetBlackboard name="enabel_yawrate" value="false" output_key="enabel_yawrate" />
        <SetBlackboard name="ctrl_type" value="0" output_key="ctrl_type" />
        <SetBlackboard name="hover_is_end" value="false" output_key="hover_is_end" />
        <SetBlackboard name="stop_hover" value="false" output_key="stop_hover" />
        <SetBlackboard name="is_time_ctrl" value="3.0" output_key="is_time_ctrl" />
        <GoalYaw goal_yaw="180" is_set_point="true" point="{P3_goal}" yaw_rate="0.0"/>
    </Sequence>
    </BehaviorTree>

  <TreeNodesModel>
    <SubTree ID="takeoff">
      <input_port name="is_rc" type="int">is remote controller operator</input_port>
      <input_port name="goal" type="BT::Position3D">Takeoff position ... </input_port>
    </SubTree>
    <SubTree ID="planner" />
    <SubTree ID="PointGoal"/>
    <SubTree ID="land" />
    <Action ID="GoalYaw">
      <input_port name="yaw_rate" type="BT::AnyTypeAllowed">use we want yaw_rate</input_port>
      <input_port name="is_set_point" type="BT::AnyTypeAllowed">if we want set point</input_port>
      <input_port name="point" type="BT::AnyTypeAllowed">set we need point,but the is_set_point is true</input_port>
      <input_port name="goal_yaw" type="BT::AnyTypeAllowed">we need to control the fcu yaw</input_port>
    </Action>
  </TreeNodesModel>

</root>